deeppavlov/models/ranking/matching_models/dam_utils/operations.py

```diff
@@ -0,0 +1,374 @@
+# @inproceedings{ ,
+#   title={Multi-Turn Response Selection for Chatbots with Deep Attention Matching Network},
+#   author={Xiangyang Zhou, Lu Li, Daxiang Dong, Yi Liu, Ying Chen, Wayne Xin Zhao, Dianhai Yu and Hua Wu},
+#   booktitle={Proceedings of the 56th Annual Meeting of the Association for Computational Linguistics (Volume 1: Long Papers)},
+#   volume={1},
+#   pages={  --  },
+#   year={2018}
+# }
+# ```
+# http://aclweb.org/anthology/P18-1103
+#
+# Based on authors' Tensorflow code: https://github.com/baidu/Dialogue/tree/master/DAM
```

Comment:
What about licencing? Is it compatible with Apache 2.0 licence? Can we use this code?

Reply:
I was not 100% sure, so I requested the information from authors. It will be clear very soon.

Reply:
1) Authors of DAM network would like to use MIT license, they think it should be compatible with Apache 2.0 license.
2) SMN license is not approved yet